lib/Widgets/IncomingTransmissionListEntry.vala

```diff
@@ -91,8 +93,7 @@ public class Drop.Widgets.IncomingTransmissionListEntry : TransmissionListEntry
         reject_button.clicked.connect (() => {
             try {
                 transmission.reject_transmission ();
-
-                accept_button.set_visible (false);
+                this.destroy ();
```

Comment:
if you clicked reject, widget is no longer needed and should disappear from the Transmission List.